typos, punctuation, small fixes

abbycross opened the pull request.

abbycross committed:
punctuation, typos, streamlining content

abbycross committed:
whitespace and typo

mtreinish commented:
Thanks for reviewing the content, most of these look llke good cleanups. But we should isolate the changes on this branch to just be the additions I made in Qiskit/qiskit#686 to keep the PR to a single logical change. The content in the file before the new `Contributing Guidelines` section should be updated in a new separate pr against Qiskit/qiskit.

mtreinish commented:
Or alternatively we can land Qiskit/Qiskit#686 as is and re-target this against Qiskit/qiskit instead as a follow up change to that file.

SooluThomas committed:
Set logo-only to False (#693)

SooluThomas committed:
unify Qiskit logo (#694)

melvin2016 committed:
Added better navigation to different elements under Installation from source section. (#691)

* added better navigation to different source installations in contributing_to_qiskit.rst file

* added with the new :ref: syntax for better navigation to different elements under Installation from source section

* prepended install to :ref: labels and reformatted to make it look like a hyperlink

* fix lint error

melvin2016 committed:
Fixed sphinx_tabs (#692)

This commit fixes #688 by removing `html_content` and adding `app.add_css_file()` instead because `html_content` was somehow blocking the css of other extensions.

mtreinish committed:
Add common contribution guidelines to docs (#686)

Right now each qiskit element has a local copy of a contributing file
which dictate the contribution guidelines for the project. However,
since Qiskit is made up of multiple elements lots of project policy
pieces are more or less the same. Although, often the guidelines get out
of sync as things get added or removed from these common pieces.

abbycross committed:
Merge remote-tracking branch 'Qiskit/master' into patch-1
